internal/bcrypt/bcrypt_windows.go

```diff
@@ -256,3 +263,13 @@ func Encrypt(hKey KEY_HANDLE, plaintext []byte, pPaddingInfo unsafe.Pointer, pbI
 //sys   DeriveKey(hSharedSecret SECRET_HANDLE, pwszKDF *uint16, pParameterList *BufferDesc, pbDerivedKey []byte, pcbResult *uint32, dwFlags uint32) (s error) = bcrypt.BCryptDeriveKey
 //sys   KeyDerivation(hKey KEY_HANDLE, pParameterList *BufferDesc, pbDerivedKey []byte, pcbResult *uint32, dwFlags uint32) (s error) = bcrypt.BCryptKeyDerivation
 //sys   DestroySecret(hSecret SECRET_HANDLE) (s error) = bcrypt.BCryptDestroySecret
+
+func GenerateSymmetricKey(hAlgorithm ALG_HANDLE, phKey *KEY_HANDLE, pbKeyObject []byte, pbSecret []byte, dwFlags uint32) error {
+	cbLen := uint32(len(pbSecret))
+	if cbLen == 0 {
+		// BCryptGenerateSymmetricKey does not support nil pbSecret,
+		// stack-allocate a zero byte here just to make CNG happy.
+		pbSecret = make([]byte, 1)
```

Comment:
Should this also set `cbLen` to `1`? Or would a zero-length slice work (as long as it isn't `nil`)?

Reply:
`0` seems more correct, as the password length is 0. Passing a non-nil buffer is just to make CNG happy.